### ISSUE TITLE
nexus-inference: safetensors loader + Python integration test infrastructure

### ISSUE THREAD

**Issue by Abso1ut3Zer0** [Abso1ut3Zer0/nexus#351]
## Context

nexus-inference models are designed to run pre-trained PyTorch models at sub-microsecond latency on CPU. The current loading story is limited to LightGBM text format for GBDT (#310). All other model types require manual weight construction — users must extract tensors themselves and pass raw slices to constructors.

The ML ecosystem has converged on [safetensors](https://github.com/huggingface/safetensors) as the standard model serialization format. It's supported natively by PyTorch, HuggingFace, and most modern ML tooling.

### 1. Safetensors loader (`loader-safetensors` feature)

Add `from_safetensors` constructors to all model types that use weight matrices:

```rust
use nexus_inference::{TinyLstmF32, MlpF32, Activation};
use nexus_inference::loader::safetensors;

let bytes = std::fs::read("model.safetensors")?;

// Prefix-based: matches real PyTorch module naming
let lstm = TinyLstmF32::from_safetensors(&bytes, "encoder.lstm")?;
let mlp = MlpF32::from_safetensors(&bytes, "encoder.fc", &[Activation::Relu])?;
```

**Design decisions:**
- **`&[u8]` input, not file paths** — caller controls I/O, `no_std` compatible (with `alloc`)
- **Prefix-based tensor resolution** — matches PyTorch's `state_dict()` naming (`encoder.lstm.weight_ih_l0`, `encoder.fc.0.weight`, etc.)
- **Activation passed as parameter** — PyTorch state_dicts contain only weights, not architecture metadata. Activation is a code-level choice.
- **GBDT keeps LightGBM loader** — tree structures don't map to tensors

**Model coverage:**

| Model | Prefix example | Tensors resolved |
|-------|---------------|-----------------|
| `TinyLstmF32` | `"lstm"` | `weight_ih_l0`, `weight_hh_l0`, `bias_ih_l0`, `bias_hh_l0` |
| `TinyGruF32` | `"gru"` | `weight_ih_l0`, `weight_hh_l0`, `bias_ih_l0`, `bias_hh_l0` |
| `MlpF32/F64` | `"fc"` | `0.weight`, `0.bias`, `1.weight`, `1.bias`, ... |
| `Causal1dConvF32` | `"conv"` | `weight`, `bias` |
| `LutF32/F64` | N/A | No weights — constructed from data, not trained |

### 2. Python integration test scripts

For each model type, a Python script that:
1. Creates a small model with deterministic weights/seed
2. Runs inference on fixed inputs
3. Saves the model via `safetensors.torch.save_file()`
4. Writes expected outputs to a companion JSON file

Scripts live in `nexus-inference/tests/fixtures/generate_*.py`. Generated `.safetensors` and `.json` files are checked into the repo (no Python dependency for `cargo test`).

### 3. Rust integration tests

Load the generated fixtures, run inference, assert outputs match Python within tolerance (1e-6 for f32, 1e-12 for f64).

```rust
#[test]
fn lstm_matches_pytorch() {
    let bytes = include_bytes!("fixtures/lstm_h16.safetensors");
    let expected: Vec<f32> = load_expected("fixtures/lstm_h16.json");
    let lstm = TinyLstmF32::from_safetensors(bytes, "lstm").unwrap();
    // ... step through inputs, compare outputs
}
```

## Expands on

- #344 (LightGBM fixtures) — same pattern, broader scope
- #310 (loaders) — safetensors is the second loader after LightGBM

## Current limitations documented

Users should be aware of constraints on the PyTorch models they can load:

- **LSTM/GRU:** Single-layer, unidirectional only (`num_layers=1`, `bidirectional=False`). See #350 for multi-layer convenience.
- **MLP:** Bias required on all layers, no LayerNorm/BatchNorm between layers. See #349 for planned enhancements.
- **Conv1d:** Single input/output channel (`in_channels=1, out_channels=1`). Multi-channel handled by TinyTcnF32 (#316).
- **Activation functions:** Must be specified in Rust code — not stored in the weights file.

**Comment by Abso1ut3Zer0** [comment]
Delivered in PR #352. Safetensors loaders for all model types (LSTM, GRU, MLP, Conv1d) with 44 PyTorch parity integration tests and CI verification.
